### DISCUSSION TITLE
MessageBox in a UserControl

### DISCUSSION THREAD

**Discussion by chasihle** [canton7/Stylet#192] [Q&A] [answered]
Hi,
I'm attempting to implement a message box in a UserControl. I can get MessageBox working in a window form with no issues but I receive "System.NullReferenceException: 'Object reference not set to an instance of an object.'"

Am I doing something wrong or can MessageBox not be implemented in a UserControl.
A code snippet is below.

Thanks

using Stylet;
using System.Collections.ObjectModel;
using System;
using System.Windows.Input;
using SmallBusCustMgtLibreary.Models;
using System.Collections.Generic;
using System.Windows;

namespace SmallBusCustMgtWPF.Pages
{
    public class CustomerNotesViewModel : Conductor<IScreen>.StackNavigation, ICommand
    {

        public event EventHandler CanExecuteChanged;

        public readonly IWindowManager windowManager;
        public MessageBoxResult result = new MessageBoxResult();

        public CustomerNotesViewModel(SystemTableModel sysmod, int custid)
        {
            this.windowManager = windowManager;
        }

        public void DeleteCustNote()
        {
            if(SelectedCustomerNote != null)
            { 
                DataAccess DA = new DataAccess();
                DA.DeleteCustNote(SelectedCustomerNote.CustNteId);
   CustomerNote = new  
   ObservableCollection<CustomerNotesModel>(DA.GetCustNoteGridRec(custId));
            }
            else
            {
                MessageBoxResult result = new MessageBoxResult();
                MessageBoxViewModel.ButtonLabels[MessageBoxResult.OK] = "OK";

                result = this.windowManager.ShowMessageBox
                        ("A Customer Note Must First Be Selected", "Customer Note Delete Failed",
                                                       buttons: MessageBoxButton.OK,
                                                       buttonLabels: new Dictionary<MessageBoxResult, string>());
            //{
            //                { MessageBoxResult.Cancel, "Cancel" },
            //                });
            }
        }

        public bool CanExecute(object parameter)
        {
            throw new NotImplementedException();
        }

        
    }
}

**Comment by canton7** [comment, marked as answer]
```
this.windowManager = windowManager
```

There's no constructor parameter called `windowManager`, so this assigns the field to itself. You need  to pass a window manager into the constructor

**Comment by chasihle** [comment]
Thanks, that works 

Sent from C. Ihle's iPhone 
704-439-6270
